packages/@uppy/core/src/Uppy.js

```diff
@@ -1386,6 +1386,22 @@ class Uppy {
     }
   }
 
+  // We need to store request clients by a unique ID, so we can share RequestClient instances across files
+  // this allows us to do rate limiting and synchronous operations like refreshing provider tokens
```

Comment:
I'm missing why we technically need to share instances across files. Conceptually it says rate limiting ans such, but that still doesn't explain to me why we need to have a global map for this 🤔 

Can we improve this for future selves?

Reply:
it's needed for refreshing tokens. if each file has their own requestclient, we don't have any way to synchronize all requests to
- block all requests
- refresh the token
- unblock all requests and allow them to run with a the new access token

back when we had a requestclient per file, once an access token expired, all 6 files would go ahead and refresh the token at the same time (calling /refresh-token up to 6 times), which will probably fail for some providers

Reply:
Can you update the comment?